src/lib/draw/StreetViewMode.svelte

```diff
@@ -1,40 +1,21 @@
 <script lang="ts">
-  import type { MapMouseEvent } from "maplibre-gl";
-  import { currentMode, map, userSettings } from "../../stores";
   import type { Mode } from "../../types";
-  import { CollapsibleCard } from "../common";
-  import { Radio } from "../govuk";
+  import StreetViewController from "../common/StreetViewController.svelte";
   import type { EventHandler } from "./event_handler";
 
   export let eventHandler: EventHandler;
 
-  const thisMode = "street-view";
-
   export let changeMode: (m: Mode) => void;
+  let streetViewController: StreetViewController;
 
   export function start() {
-    $map.getCanvas().style.cursor = "zoom-in";
+    eventHandler.mapHandlers.click = streetViewController.handleMapClickEvent;
+    streetViewController.enableStreetView();
   }
   export function stop() {
-    $map.getCanvas().style.cursor = "inherit";
+    streetViewController.disableStreetView();
   }
 
-  eventHandler.mapHandlers.click = (e: MapMouseEvent) => {
-    let lon = e.lngLat.lng;
-    let lat = e.lngLat.lat;
-    if ($userSettings.streetViewImagery == "google") {
-      window.open(
-        `http://maps.google.com/maps?q=&layer=c&cbll=${lat},${lon}&cbp=11,0,0,0,0`,
-        "_blank"
-      );
-    } else if ($userSettings.streetViewImagery == "bing") {
-      window.open(
-        `https://www.bing.com/maps?cp=${lat}~${lon}&style=x`,
-        "_blank"
-      );
-    }
-  };
-
   eventHandler.documentHandlers.keydown = (e: KeyboardEvent) => {
```

Comment:
We could consider using the escape key on the browse page too, but no need for it now